expression/builtin_miscellaneous.go

```diff
@@ -550,28 +579,8 @@ func (b *builtinIsIPv4CompatSig) eval(row []types.Datum) (d types.Datum, err err
 		return d, errors.Trace(err)
 	}
 
-	arg := args[0]
-	d.SetInt64(0)
-	if arg.IsNull() {
-		return d, nil
-	}
-
-	ipAddress, err := arg.ToBytes()
-	if err != nil {
-		return d, errors.Trace(err)
-	}
-
-	if len(ipAddress) != net.IPv6len {
-		//Not an IPv6 address, return false
-		return
-	}
-
 	var v4InV6Prefix = []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
-	if !bytes.HasPrefix(ipAddress, v4InV6Prefix) {
-		return
-	}
-
-	d.SetInt64(1)
+	d, err = checkIPv6Prefix(&args, &v4InV6Prefix)
```

Comment:
Why not add a attribute in signature class named prefix and share the signature between builtinIsIPv4Compat and builtinIsIPv4Mapped?